vcf_annotation_tools/transform_split_values.py

```diff
@@ -0,0 +1,198 @@
+    for operation in args.operations:
+        if operation in ['ref', 'alt', 'ref_ratio', 'alt_ratio'] and format_field_info.number != 'R':
+            vcf_reader.close()
+            raise Exception("ERROR: VCF {} format field {} incompatible with operation {}. Not of Number R.".format(args.input_vcf, args.format_field, operation))
+        if operation in ['sum', 'min', 'max', 'mean', 'median', 'stdev', 'ref_ratio', 'alt_ratio'] and format_field_info.type not in ['Integer', 'Float']:
+            vcf_reader.close()
+            raise Exception("ERROR: VCF {} format field {} incompatible with operation {}. Not of Type Integer or Float.".format(args.input_vcf, args.format_field, operation))
+    return vcf_reader, sample_name
+
+def create_tsv_reader(input_filehandle):
+    tsv_reader = csv.DictReader(input_filehandle, delimiter = "\t")
+    for field in ['CHROM', 'POS', 'REF', 'ALT']:
+        if field not in tsv_reader.fieldnames:
+            raise Exception("ERROR: Input TSV {} doesn't contain required column '{}'.".format(input_filehandle.name, field))
+    return tsv_reader
+
+def ref(field_values):
+    return handle_empty_array(field_values, "field_values[0]")
+
+def alt(field_values):
+    return handle_empty_array(field_values, "field_values[1]")
+
+def ref_ratio(field_values):
+    return handle_empty_array(field_values, "field_values[0] / sum(field_values)")
+
+def alt_ratio(field_values):
+    return handle_empty_array(field_values, "field_values[1] / sum(field_values)")
+
+def handle_empty_array(field_values, action):
+    if len(field_values) == 0:
+        return '-'
+    else:
+        return eval(action)
+
+def extract_field_values(args):
+    (vcf_reader, sample_name) = create_vcf_reader(args)
+    values = {}
+    for variant in vcf_reader:
+        chr = str(variant.CHROM)
+        pos = str(variant.POS)
+        reference = str(variant.REF)
+        alts = variant.ALT
+
+        if chr not in values:
+            values[chr] = {}
+        if pos not in values[chr]:
+            values[chr][pos] = {}
+        if reference not in values[chr][pos]:
+            values[chr][pos][reference] = {}
+        alts_string = ','.join(a.serialize() for a in alts)
+        if alts_string not in values[chr][pos][reference]:
+            values[chr][pos][reference][alts_string] = {}
+
+        for operation in args.operations:
+            if operation in ['ref', 'alt', 'ref_ratio', 'alt_ratio'] and len(alts) > 1:
+                print("Multi-allelic sites are not supported for operation {}. Skipping entry {} {}.".format(operation, chr, pos))
+                values[chr][pos][reference][alts_string][operation] = '-'
+                continue
+
+            if args.format_field in variant.call_for_sample[sample_name].data:
+                field_value = variant.call_for_sample[sample_name].data[args.format_field]
+                result = eval("{}({})".format(operation, field_value))
```

Comment:
This is where my lack of python knowledge shines, but would `getattr` be a way to avoid an `eval` here?

Reply:
That won't work quite as straightforward since some operations are methods in this module and some come from the statistics module. So then we'd need to implement local methods for all of them that would call out to the statistics version under the hood.